.github/workflows/ci.yml

```diff
@@ -0,0 +1,409 @@
+        name: Generate nginx
+        run: |
+          echo "::set-output name=versions::[\"1.17-nginx1-nginx\", \"1.16\", \"1.15\", \"1.14\"]"
+  generate-php-strategy:
+    name: Generate PHP
+    runs-on: ubuntu-latest
+    outputs:
+      php: ${{ steps.generate-php-strategy.outputs.versions }}
+    steps:
+      - id: generate-php-strategy
+        name: Generate PHP
+        run: |
+          echo "::set-output name=versions::[\"7.4\", \"7.3\", \"7.2\"]"
+  generate-php-type-strategy:
+    name: Generate PHP Type
+    runs-on: ubuntu-latest
+    outputs:
+      type: ${{ steps.generate-php-type-strategy.outputs.type }}
+    steps:
+      - id: generate-php-type-strategy
+        name: Generate Type
+        run: |
+          echo "::set-output name=type::[\"cli\", \"fpm\"]"
+  generate-type-strategy:
+    name: Generate Type
+    runs-on: ubuntu-latest
+    needs:
+      - generate-php-type-strategy
+    outputs:
+      type: ${{ steps.generate-type-strategy.outputs.type }}
+    steps:
+      - id: generate-type-strategy
+        name: Generate Type
+        run: |
+          echo "::set-output name=type::[\"cli\", \"fpm\", \"http\"]"
+  lint-docker:
+    name: Lint Dockerfile-${{ matrix.type }}
+    runs-on: ubuntu-latest
+    needs:
+      - generate-type-strategy
+    strategy:
+      fail-fast: false
+      matrix:
+        type: ${{ fromJson(needs.generate-type-strategy.outputs.type) }}
+    steps:
+      - uses: actions/checkout@v2
+      - name: Lint Dockerfile-${{ matrix.type }}
+        uses: docker://hadolint/hadolint:latest-debian
+        with:
+          entrypoint: hadolint
+          args: Dockerfile-${{ matrix.type }}
+  lint-shell:
+    name: Lint shell scripts
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+    steps:
+      - uses: actions/checkout@v2
+      - id: files
+        name: Generate shell script file list
+        run: |
+          list=$(ls src/http/nginx/docker* src/php/utils/install-* src/php/utils/docker/* build* test-*  | tr "\n" " " | sed -z '$ s/\n$//')
+          echo -e "::set-output name=list::/github/workspace/${list// / /github/workspace/}build-http.sh"
+      - name: Lint shell scripts
+        uses: docker://koalaman/shellcheck:latest
+        with:
+          args: ${{ steps.files.outputs.list }}
+  build-nginx:
+    name: Build nginx ${{ matrix.nginx }}
+    needs:
+      - lint-docker
+      - lint-shell
+      - generate-nginx-strategy
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        nginx: ${{ fromJson(needs.generate-nginx-strategy.outputs.nginx) }}
+    steps:
+      - uses: actions/checkout@v2
+      - run: ./build-http.sh $(echo "${{ matrix.nginx }}" | tr '-' ' ')
+      - run: cat ./tmp/image.tags | xargs -I % docker inspect --format='%={{.Id}}:{{index .Config.Env 7}}' %
+      - run: docker save "${DOCKER_IMAGE}" | gzip -9 > ./tmp/image-http-${{ matrix.nginx }}.tar
+      - run: docker images
```

Comment:
I don't think need to list all the images on the host, could it be a leftover debug line? (same goes for the other `build-*` and `push-*` jobs)

Reply:
Indeed, looks like it's a debug line or something: https://github.com/usabilla/php-docker-template/runs/1309455144?check_suite_focus=true#step:6:1

I'll remove it, thanks!